### ISSUE TITLE
Conditional localized blocks result in database error

### ISSUE THREAD

**Issue by abel-sch** [payloadcms/payload-3.0-demo#121]
This is a bit of a edge case here:
Creating a different blocks field based on another select field with localization results in collections page ('pages' in this example) to crash with the following error:

```
 ⨯ Internal error: TypeError: Cannot read properties of undefined (reading 'referencedTable')
    at normalizeRelation (./node_modules/drizzle-orm/relations.js:211:56)
    at PgDialect.buildRelationalQueryWithoutPK (./node_modules/drizzle-orm/pg-core/dialect.js:963:101)
    at QueryPromise._getQuery (./node_modules/drizzle-orm/pg-core/query-builders/query.js:90:25)
    at QueryPromise._toSQL (./node_modules/drizzle-orm/pg-core/query-builders/query.js:105:24)
    at eval (./node_modules/drizzle-orm/pg-core/query-builders/query.js:69:42)
    at Object.startActiveSpan (./node_modules/drizzle-orm/tracing.js:14:14)
    at QueryPromise._prepare (./node_modules/drizzle-orm/pg-core/query-builders/query.js:68:60)
    at eval (./node_modules/drizzle-orm/pg-core/query-builders/query.js:114:19)
    at Object.startActiveSpan (./node_modules/drizzle-orm/tracing.js:14:14)
    at QueryPromise.execute (./node_modules/drizzle-orm/pg-core/query-builders/query.js:113:60)
    at QueryPromise.then (./node_modules/drizzle-orm/query-promise.js:26:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
 
```

I think there's something wrong with the database schema. I get it to work when I either remove localization or the conditional logic to the blocks fields.

This is what my schema looks like:
```
{
   "slug":"pages",
   "fields":[
      {
         "name":"title",
         "label":"Title",
         "type":"text",
         "required":true,
         "localized":true
      },
      {
         "name":"template",
         "type":"select",
         "options":[
            {
               "label":"Default",
               "value":"Default"
            },
            {
               "label":"Home",
               "value":"Home"
            }
         ],
         "defaultValue":"Default",
         "admin":{
            "position":"sidebar"
         }
      },
      {
         "name":"DefaultLayout",
         "type":"blocks",
         "localized": true,
         "blocks":[
            {
               "slug":"Quote",
               "fields":[ ... ]
            },
            {
               "slug":"Slider",
               "fields":[ ... ]
            }
         ],
         "admin":{
            condition: (data) => {
               return data.template === 'Default'
           }
         }
      },
      {
         "name":"HomeLayout",
         "type":"blocks",
         "localized": true,
         "blocks":[
            {
               "slug":"Quote",
               "fields":[ ... ]
            }
         ],
         "admin":{
            condition: (data) => {
               return data.template === 'Home'
           }
         }
      }
   ]
}

**Comment by abel-sch** [comment]
I have managed to get this _working_ by updating to beta11 however, when running for a bit and clicking around in the cms, and making calls to the local AP (without making any changes to the db) I get these same Drizzle errors. Restarting the dev server makes it work for a bit again. This makes me think it is not related to localization or blocks per se but that postgres / drizzle implementation is unstable.

PS: all table / columns are well under 64 chars

**Comment by DanRibbens** [comment]
I believe this is from foreign key names becoming too long which was fixed by this PR: https://github.com/payloadcms/payload/pull/5976

This was just released in [3.0.0-beta.14](https://www.npmjs.com/package/@payloadcms/db-postgres/v/3.0.0-beta.14)

**Comment by DanRibbens** [comment]
Please tag me and I will reopen this issue if this doesn't resolve it for you. Thanks!
